src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -108,58 +122,95 @@ public PaymentHandleResponse patchAddress(@Nonnull Cart cart, @Nonnull String pa
             Payment paypalPlusPayment = new Payment().setId(paypalPlusPaymentId);
             ShippingAddress shippingAddress = shippingAddressMapper.ctpAddressToPaypalPlusAddress(cart);
             Patch replace = new Patch("add", "/transactions/0/item_list/shipping_address").setValue(shippingAddress);
-            return paypalPlusFacade.getPaymentService().patch(paypalPlusPayment, replace)
-                    .thenApply(payment -> PaymentHandleResponse.ofStatusCode(HttpStatus.OK))
-                    .exceptionally(throwable -> {
-                        logger.error("Unexpected exception handling payment cartId=[{}]:", cart.getId(), throwable);
-                        return PaymentHandleResponse.of400BadRequest(
-                                format("Payment or cart for cartId==[%s] can't be processed, see the logs", cart.getId()));
-                    })
-                    .toCompletableFuture().join();
+            CompletionStage<PaymentHandleResponse> patchCS = paypalPlusFacade.getPaymentService().patch(paypalPlusPayment, replace)
+                    .thenApply(payment -> PaymentHandleResponse.ofStatusCode(HttpStatus.OK));
+            return paymentExceptionWrapper(paypalPlusPaymentId, patchCS);
         } catch (Exception e) {
```

Comment:
Yes, but in `paymentExceptionWrapper()` it just catches CompletionStage exceptions. There could be errors e.g. in the mapper, which will be catch in try/catch.